godot/src/ui/components/place_item.gd

```diff
@@ -0,0 +1,279 @@
+
+
+func _get_label_likes() -> Label:
+	return _get_node_safe("Label_Likes")
+
+
+func _get_container_likes() -> Control:
+	return _get_node_safe("HBoxContainer_Likes")
+
+
+func _get_texture_image() -> TextureRect:
+	return _get_node_safe("TextureRect_Image")
+
+
+func _connect_signals():
+	var button_close = _get_button_close()
+	if button_close:
+		if not button_close.pressed.is_connected(_on_button_close_pressed):
+			button_close.pressed.connect(_on_button_close_pressed)
+
+	var button_jump_in = _get_button_jump_in()
+	if button_jump_in:
+		if not button_jump_in.pressed.is_connected(_on_button_jump_in_pressed):
+			button_jump_in.pressed.connect(_on_button_jump_in_pressed)
+
+	var button_like = _get_button_like()
+	if button_like:
+		if not button_like.pressed.is_connected(_on_button_like_pressed):
+			button_like.pressed.connect(_on_button_like_pressed)
+
+
+func set_location(_location: Vector2i):
+	var label = _get_label_location()
+	if label:
+		location = _location
+		label.text = "%s, %s" % [_location.x, _location.y]
+
+
+func set_image(_texture: Texture2D):
+	var texture_rect = _get_texture_image()
+	if texture_rect:
+		texture_rect.texture = _texture
+
+
+func set_title(_title: String):
+	var label = _get_label_title()
+	if label:
+		label.text = _title
+
+
+func set_description(_description: String):
+	var label = _get_label_description()
+	if label:
+		label.text = _description
+
+
+func set_views(_views: int):
+	var label = _get_label_views()
+	var container = _get_container_views()
+	if label and container:
+		container.set_visible(_views > 0)
+		label.text = _format_number(_views)
+
+
+func set_likes_percent(_likes: float):
+	var label = _get_label_likes()
+	var container = _get_container_likes()
+	if label and container:
+		container.set_visible(_likes > 0.0)
+		label.text = str(round(_likes * 100)) + "%"
+
+
+func set_online(_online: int):
+	var label = _get_label_online()
+	var container = _get_container_online()
+	if label and container:
+		container.set_visible(_online > 0)
+		label.text = _format_number(_online)
+
+
+func set_realm(_realm: String, _realm_title: String):
+	realm = _realm
+	var label = _get_label_realm()
+	if label:
+		label.text = _realm_title
+
+
+func set_creator(_creator: String):
+	var label = _get_label_creator()
+	var container = _get_container_creator()
+	if label and container:
+		container.visible = not _creator.is_empty()
+		label.text = _creator
+
+
+func set_data(item_data):
+	_data = item_data
+	set_title(item_data.get("title", ""))
+	set_description(_get_or_empty_string(item_data, "description"))
+
+	set_views(item_data.get("user_visits", 0))
+	var like_score = item_data.get("like_score", 0.0)
+	set_likes_percent(like_score if like_score is float else 0.0)
+	set_online(item_data.get("user_count", 0))
+
+	var image_url = item_data.get("image", "")
+	if not image_url.is_empty():
+		set_image(texture_placeholder)
+		_async_download_image(image_url)
+	else:
+		set_image(texture_placeholder)
```

Comment:
check if `_get_texture_image` exists... if not, don't try to _async_download_image... you can put all the code above inside the if (no need to put the placeholder also)...